newIDE/app/src/Profile/Team/TeamProvider.js

```diff
@@ -0,0 +1,224 @@
+// @flow
+
+import * as React from 'react';
+import TeamContext from './TeamContext';
+import {
+  listTeamGroups,
+  listTeamMembers,
+  listTeamMemberships,
+  listUserTeams,
+  updateGroup,
+  type Team,
+  type TeamGroup,
+  type TeamMembership,
+  type User,
+  updateUserGroup,
+  deleteGroup,
+  createGroup,
+} from '../../Utils/GDevelopServices/User';
+import AuthenticatedUserContext from '../../Profile/AuthenticatedUserContext';
+import { listOtherUserCloudProjects } from '../../Utils/GDevelopServices/Project';
+
+type Props = {| children: React.Node |};
+
+const TeamProvider = ({ children }: Props) => {
+  const { profile, getAuthorizationHeader, authenticated } = React.useContext(
+    AuthenticatedUserContext
+  );
+  const [groups, setGroups] = React.useState<?(TeamGroup[])>(null);
+  const [team, setTeam] = React.useState<?Team>(null);
+  const [members, setMembers] = React.useState<?(User[])>(null);
+  const [memberships, setMemberships] = React.useState<?(TeamMembership[])>(
+    null
+  );
+
+  React.useEffect(
+    () => {
+      if (!authenticated) {
+        setTeam(null);
+        setGroups(null);
+        setMembers(null);
+        setMemberships(null);
+      }
+    },
+    [authenticated]
+  );
+
+  React.useEffect(
+    () => {
+      const fetchTeam = async () => {
+        if (!profile || !profile.isTeacher) return;
+        const teams = await listUserTeams(getAuthorizationHeader, profile.id);
+        // Being admin of multiple teams is not supported at the moment.
```

Comment:
We'll make something to switch between teams in the future (a selector somewhere) :)